### PR TITLE
Envelope action service

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSBPS-363

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/EnvelopeActionService.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        if (envelope.status == DISPATCHED || envelope.status == REJECTED
+            || !isStale(envelope)) {
+            throw new EnvelopeCompletedOrNotStaleException(
+                "Envelope with id " + envelope.id + " is completed or not stale"
+            );
+        }
+    }
+
+    private boolean isStale(Envelope envelope) {
+        if (envelope.status != Status.CREATED) {
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/EnvelopeActionService.java:75]: addition to this, if it is CFT envelope can we check processor?

Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/EnvelopeActionService.java:75]: Processor and router have different databases. As mentioned in https://tools.hmcts.net/jira/browse/DTSBPS-348?focusedCommentId=952256&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-952256 there are several zip files in router which are not known by processor. This means that we need to check and handle in processor separately.

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/EnvelopeActionService.java:75]: there is an end point to check

Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/EnvelopeActionService.java:75]: The idea here is to handle an envelope which is stale in router i.e. in CREATED status with after e.g. 2 hours. It does not matter if it exists in processor.

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/EnvelopeActionService.java:75]: if file is processed in processor successfully you can not mark as rejected?

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/events/EventType.java [hunk 2]
```diff
@@ -9,4 +9,6 @@
     DUPLICATE_REJECTED,
     ERROR,
     NOTIFICATION_SENT,
+    MANUALLY_MARKED_AS_REJECTED,
+    MANUALLY_MARKED_AS_DISPATCHED
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/data/events/EventType.java:13]: do we use event type in any report or would it affect any reports?

Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/data/events/EventType.java:13]: In this report
image

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/data/events/EventType.java:13]: I mean if we are classifying with event type that would affect the report result.

**Review by mdayican — APPROVED** [review]

**Pull request merged**